### ISSUE TITLE
[Question] How to define multiple assign statements?

### ISSUE THREAD

**Issue by log1-c** [telekom-mms/ansible-collection-icinga-director#226]
### Question

Hello :)
First of all, thanks for the great module!

I'd like to ask how to define multiple assign statements so that I end up with something like this:
![image](https://github.com/telekom-mms/ansible-collection-icinga-director/assets/24474580/83bdd140-0c8d-48d1-83b0-1f36b52ad5b9)
`assign where host.vars.hostgroup == "msd" || host.vars.tags.customer == "msd"`

I have tried the following:
`assign_filter: "host.vars.hostgroup=%22{{ var_pb_customer_token }}%22%7Chost.vars.tags.customer=%22{{ var_pb_customer_token }}%22"` (also tried with a literal `|`, `or` and `||`)
which leads to
`/** Failed to render this object **/
/*  Got invalid JSON in filter string: host.vars.hostgroup="q1au2"|host.vars.tags.customer="q1au2" */`

I have also tried the part from the config preview
`host.vars.hostgroup == "msd" || host.vars.tags.customer == "msd"`
this also leads to `/** Failed to render this object **/
/*  Got invalid JSON in filter string: `

A single condition like `assign_filter: "host.vars.hostgroup=%22{{ var_pb_customer_token }}%22"` works without problems.

If it makes a difference I'm trying to create a host _group_.

🙋‍♂️

**Comment by rndmh3ro** [comment]
I have locally defined this assign_filter:
```
assign_filter: host.name="{{ "*customer-prod*" }}" | host.name="customer-azure"
```

This results in:

```
> curl  -H "Accept: application/json" https://example.com/director/hostGroup?uuid=821b02b9-874c-49c4-981b-219000000006
{
    "assign_filter": "host.name=\"*customer-prod*\" | host.name=\"customer-azure\"",
    "object_name": "customer-prod",
    "object_type": "object"
}
```

![image](https://github.com/telekom-mms/ansible-collection-icinga-director/assets/3198961/da87a237-d15a-4c2b-b05b-65616b18c44b)

Can you try this?

**Comment by log1-c** [comment]
Well, that is kinda confusing what the Director does here...

**Having multiple conditions does work like this**
```
director/hostgroup?name=q1poc -X POST -d '{"assign_filter": "host.vars.hostgroup=\"q1poc\"|host.vars.tags.customer=\"q1poc\"", "object_type": "object", "object_name": "q1poc"}'
```
![image](https://github.com/telekom-mms/ansible-collection-icinga-director/assets/24474580/1eff03ec-cb5d-4819-b9cc-d7dc395147e5)
```
{
    "assign_filter": "host.vars.hostgroup=\"q1poc\"|host.vars.tags.customer=\"q1poc\"",
    "object_name": "q1poc",
    "object_type": "object"
}
```

**but not with URL-encoding**
```
/director/hostgroup?name=q1poc -X POST -d '{"assign_filter": "host.vars.hostgroup=%22q1poc%22%7Chost.vars.tags.customer=%22q1poc%22", "object_type": "object", "object_name": "q1poc"}'
```
![image](https://github.com/telekom-mms/ansible-collection-icinga-director/assets/24474580/9e697651-53e4-4b13-85ff-90b8966f6196)
```
{
    "assign_filter": "host.vars.hostgroup=%22q1poc%22%7Chost.vars.tags.customer=%22q1poc%22",
    "object_name": "q1poc",
    "object_type": "object"
}
```
**while with a single condition it doesn't care about the URL encoding**
```
/director/hostgroup?name=q1poc -X POST -d '{"assign_filter": "host.vars.hostgroup=%22q1poc%22", "object_type": "object", "object_name": "q1poc"}'
```
![image](https://github.com/telekom-mms/ansible-collection-icinga-director/assets/24474580/d8a040c5-7e29-44a8-a386-3516e980b3ea)
```
{
    "assign_filter": "host.vars.hostgroup=%22q1poc%22",
    "object_name": "q1poc",
    "object_type": "object",
    "uuid": "5cd672fb-71e4-4c9d-93ae-da57c5491976"
}
```

Will try this now with Ansible and the module. If that works without problems as well (which I expect), I'll close the issue.
Thanks for pointing me in the right direction!

**Comment by log1-c** [comment]
As expected it works via the Ansible module as well, when using the above mentioned first format of the condition.
Small request: Such an "advanced" filter should also be mentioned in the docs/examples. It was a bit of a journey getting to the point at which I knew what format the module expected.

**Comment by rndmh3ro** [comment]
> Such an "advanced" filter should also be mentioned in the docs/examples

Good idea!

do you want to create a PR to add an example [here](https://github.com/telekom-mms/ansible-collection-icinga-director/blob/master/plugins/modules/icinga_hostgroup.py#L80)?

You can check the [Contributing Guidelines](https://github.com/telekom-mms/ansible-collection-icinga-director/blob/master/CONTRIBUTING.md) if you need help.

**Comment by log1-c** [comment]
Done :)
https://github.com/telekom-mms/ansible-collection-icinga-director/pull/227

Also added a section to the general read-me.

**Comment by rndmh3ro** [comment]
fixed in #227
